docs/diagrams/AddCommandHistory1.puml

```diff
@@ -0,0 +1,26 @@
+@startuml
+!include style.puml
+skinparam ClassFontColor #000000
+skinparam ClassBorderColor #000000
+skinparam ClassBackgroundColor #FFFFAA
+
+title After adding new command
+
+package Commands <<rectangle>> {
+    class Command0 as "list\n<font size=10><color:#555555>0</color></font>"
+    class Command1 as "find roy\n<font size=10><color:#555555>1</color></font>"
+    class Command2 as "add...\n<font size=10><color:#555555>2</color></font>"
+    class Command3 as "delete 5\n<font size=10><color:#555555>3</color></font>"
+    class Command4 as "list\n<font size=10><color:#555555>4</color></font>"
+}
+Command0 -[hidden]right-> Command1
+Command1 -[hidden]right-> Command2
+Command2 -[hidden]right-> Command3
+Command3 -[hidden]right-> Command4
+
+hide Command4
+
+class Pointer as "currentCommandIndex" #FFFFFF
+
+Pointer -up-> Command4
```

Comment:
I think after adding 5 commands as this diagram suggests, the `currentCommandIndex` becomes 5 instead of 4 (not sure how this would be illustrated on a diagram)? Might be because of the previously mentioned issue that `currentCommandIndex` is currently set to size.

Reply:
yea im planning to fix that after clarifying. Might just point it to an empty box to show that its ready to accept the next new command I guess

Reply:
Ohh I see, sounds good